pkg/driver/node.go

```diff
@@ -73,12 +93,41 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 	// TODO when CreateVolume is implemented, it must use the same key names
 	subpath := "/"
 	encryptInTransit := true
+	iamAuth := false
 	volContext := req.GetVolumeContext()
+
 	for k, v := range volContext {
+
 		switch strings.ToLower(k) {
+		case "csi.storage.k8s.io/pod.name":
+			continue
+		case "csi.storage.k8s.io/pod.namespace":
+			continue
+		case "csi.storage.k8s.io/pod.uid":
+			continue
+		case "csi.storage.k8s.io/serviceaccount.name":
+			continue
+		case "csi.storage.k8s.io/serviceaccount.tokens":
+			//TODO get the token in a cleaner way, not like this.
+			result := make(map[string]interface{})
+			json.Unmarshal([]byte(volContext["csi.storage.k8s.io/serviceAccount.tokens"]), &result)
+			tokenInfo := fmt.Sprintf("%v", result["sts.amazonaws.com"].(map[string]interface{})["token"])
+
+			// Set the podauth object then call set pod credentials (ROLE-ARN and Toke file) that are called through efs-utils
+			podauth := newPodAuth(ctx, volContext["csi.storage.k8s.io/pod.namespace"], volContext["csi.storage.k8s.io/serviceAccount.name"], d.k8sClient, tokenInfo)
+
+			podauth.setPodCredentials()
+		case "iamauth":
+			var err error
+			iamAuth, err = strconv.ParseBool(v)
+			if err != nil {
+				return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Volume context property %q must be a boolean value: %v", k, err))
+			}
+		case "csi.storage.k8s.io/ephemeral":
+			continue
```

Comment:
You could collapse this into the case above as well

Reply:
This is not part of my change, it should be kept as is not merged with the other CSIDriver parameters.